modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/serializer/ComponentDataSerializerProvider.java

```diff
@@ -1,34 +1,56 @@
 package com.enonic.xp.core.impl.content.serializer;
 
+import java.util.Optional;
+
 import com.enonic.xp.region.ComponentType;
 import com.enonic.xp.region.FragmentComponentType;
 import com.enonic.xp.region.ImageComponentType;
 import com.enonic.xp.region.LayoutComponentType;
+import com.enonic.xp.region.LayoutDescriptorService;
 import com.enonic.xp.region.PartComponentType;
 import com.enonic.xp.region.TextComponentType;
 
-public final class ComponentDataSerializerProvider
+public class ComponentDataSerializerProvider
 {
-    private final PartComponentDataSerializer partDataSerializer;
+    protected final PartComponentDataSerializer partDataSerializer;
+
+    protected final TextComponentDataSerializer textDataSerializer;
 
-    private final TextComponentDataSerializer textDataSerializer;
+    protected LayoutComponentDataSerializer layoutDataSerializer;
 
-    private final LayoutComponentDataSerializer layoutDataSerializer;
+    protected final ImageComponentDataSerializer imageDataSerializer;
 
-    private final ImageComponentDataSerializer imageDataSerializer;
+    protected final FragmentComponentDataSerializer fragmentDataSerializer;
 
-    private final FragmentComponentDataSerializer fragmentDataSerializer;
+    protected final RegionDataSerializer regionDataSerializer;
 
-    private final RegionDataSerializer regionDataSerializer;
+    public ComponentDataSerializerProvider()
+    {
+        this( Optional.empty() );
+    }
+
+    protected ComponentDataSerializerProvider( final LayoutDescriptorService layoutDescriptorService )
+    {
+        this( Optional.ofNullable( layoutDescriptorService ) );
+    }
 
-    public ComponentDataSerializerProvider( )
+    private ComponentDataSerializerProvider( final Optional<LayoutDescriptorService> optionalLayoutDescriptorService )
```

Comment:
probably can be simplified without need to use Optional. null check works as good